tests/Endpoints/TransactionTest.php

```diff
@@ -128,6 +133,67 @@ public function getInvalidDataScenarios(): iterable
         ];
     }
 
+    /**
+     * Test getters on transaction.
+     *
+     * @return void
+     */
+    public function testGetters(): void
+    {
+        $allocation = new Allocation();
+        $amount = new Amount([]);
+        $fundingSource = new PaymentSource([]);
+        $parent = new Transaction([]);
+        $destination = new PaymentSource([]);
+        $source = new PaymentSource([]);
+        $security = new Security([]);
+        $user = new User([]);
+
+        $entity = new Transaction([
+            'action' => '1',
+            'allocation' => $allocation,
+            'amount' => $amount,
+            'approved' => true,
+            'createdAt' => '2010-10-10T00:00:00Z',
+            'description' => 'short description',
+            'finalisedAt' => '2010-10-10T00:00:02Z',
+            'fundingSource' => $fundingSource,
+            'metadata' => ['1' => '2'],
+            'parent' => $parent,
+            'paymentDestination' => $destination,
+            'paymentSource' => $source,
+            'response' => ['ok' => 'done'],
+            'security' => $security,
+            'state' => 1,
+            'statementDescription' => 'statement description',
+            'status' => 'pending',
+            'transactionId' => 'transaction-id',
+            'updatedAt' => '2010-11-10T00:00:00Z',
+            'user' => $user,
+        ]);
+
+        self::assertSame('1', $entity->getAction());
+        self::assertSame($allocation, $entity->getAllocation());
+        self::assertSame($amount, $entity->getAmount());
+        self::assertTrue($entity->getApproved());
+        self::assertSame('2010-10-10T00:00:00Z', $entity->getCreatedAt());
+        self::assertSame('short description', $entity->getDescription());
+        self::assertSame('2010-10-10T00:00:02Z', $entity->getFinalisedAt());
+        self::assertSame($fundingSource, $entity->getFundingSource());
+        self::assertSame(['1' => '2'], $entity->getMetadata());
+        self::assertSame($parent, $entity->getParent());
+        self::assertSame($destination, $entity->getPaymentDestination());
+        self::assertSame($source, $entity->getPaymentSource());
+        self::assertSame(['ok' => 'done'], $entity->getResponse());
+        self::assertSame($security, $entity->getSecurity());
+        self::assertSame(1, $entity->getState());
```

Comment:
Minor nitpick. You should avoid using the same values for two different values being tested (in this case, action & state). If there's a copy paste error like so; `function getAction(): int { return $this->state; }` then then the tests won't pick this up.

Reply:
fixed.